<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- TODO: set in parameters somewhere
    Device map:

    /dev/ttyACM0 - Teensyduino_USB_Serial_11263870   -> teensy usb -> head
    /dev/ttyACM1 - Teensyduino_USB_Serial_8880670   -> teensy usb -> tracks
    /dev/ttyUSB0 - FTDI_Quad_RS232-HS               -> teensy serial -> tracks
    /dev/ttyUSB1 - FTDI_Quad_RS232-HS               -> arduino serial -> middle
    /dev/ttyUSB2 - FTDI_Quad_RS232-HS               
    /dev/ttyUSB3 - FTDI_Quad_RS232-HS               -> teensy serial -> head
    /dev/ttyUSB4 - 1a86_USB2.0-Serial               -> arduino usb -> middle
    
    -->

    <!-- HEAD =========================================================================== -->
    <node name="clare_head" pkg="rosserial_python" type="serial_node.py">
        <param name="port" value="/dev/ttyUSB3"/>
        <param name="baud" value="115200"/>
    </node>
    
    <node name="clare_head_camera" pkg="clare_head_camera" type="face_detect_node.py">
        <param name="model_dir" value="/home/dgorissen/openvino_models/intel/" type="str" />
        <param name="model_name" value="face-detection-retail-0004" type="str" />
        <param name="threshold" value="0.9" type="double" />
        <param name="parallel_requests" value="4" type="int" />
    </node>

    <node name="clare_respeaker" pkg="respeaker_ros" type="respeaker_node.py"/>

    <node name="clare_respeaker_stt" pkg="respeaker_ros" type="speech_to_text.py">
        <remap from="/audio" to="/speech_audio"/>
        <param name="self_cancellation" value="False" type="bool" />
        <param name="language" value="en-GB" type="str"/>
    </node>

    <include file="$(find realsense2_camera)/launch/rs_camera.launch"/>

    <!-- BODY MIDDLE ====================================================================== -->
    <node name="clare_middle" pkg="clare_middle" type="middle_node.py">
        <!-- Both the serial and usb outputs can be used -->
        <param name="port" value="/dev/ttyUSB1"/>
        <param name="baud" value="115200"/>
    </node>

    <!-- BODY TOP ========================================================================= -->
    <node name="clare_ir" pkg="clare_ir" type="ir_node.py"/>    
    <node name="clare_buttons" pkg="clare_buttons" type="buttons_node.py"/>
    <node name="clare_arms" pkg="clare_arms" type="arms_node.py">
        <param name="i2c_bus" value="3"/>
    </node>
    <node name="clare_neck" pkg="clare_neck" type="neck_node.py">
        <param name="i2c_bus" value="3"/>
    </node>
    <node name="clare_fan" pkg="clare_fan" type="fan_node.py"/>
    <node name="clare_env" pkg="clare_env" type="env_node.py">
        <param name="i2c_bus" value="3"/>
    </node>
    <node name="clare_lightring" pkg="clare_lightring" type="lightring_node.py"/>
    <node name="clare_tts" pkg="clare_tts" type="tts_node.py">
        <env name="ALSA_CARD" value="1"/>
        <param name="backend" value="gtts"/>
    </node>

    <!-- TRACKS =========================================================================== -->
    <node name="clare_tracks" pkg="rosserial_python" type="serial_node.py">
        <param name="port" value="/dev/ttyUSB0"/>
        <param name="baud" value="115200"/>
    </node>

    <!-- CONTROLLER ===================================================================== -->
    <node name="clare_controller" pkg="clare_controller" type="controller_node.py">
    </node>

</launch>
